### PR TITLE
Offer Docker image, fix Teiid typos

### PR DESCRIPTION
**Pull request by emmanuelbernard**

### CODE REVIEW

File: bin/docker/README.adoc [hunk 1]
```diff
@@ -0,0 +1,136 @@
+= Running infinispan.github.io within Docker
+
+The following instructions allow you to run hibernate.org within
```
Comment (rvansa) [inline bin/docker/README.adoc]: hibernate.org?

Comment (emmanuelbernard) [inline bin/docker/README.adoc]: I've fixed the incorrect reference and forced push.

File: bin/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
+FROM       fedora:22
+
+# install the required dependencies to compile native extensions
+RUN        dnf -y install gcc-c++ make ruby-devel libxml2-devel libxslt-devel findutils git ruby
+RUN        dnf -y install wget # needed by the website to fetch resources
```
Comment (gustavonalle) [inline bin/docker/Dockerfile]: Those two ```RUN``` could be collapsed in a single line to minimize number of layers

File: bin/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+RUN        groupadd -r dev && useradd  -g dev -u 1000 dev
+RUN        mkdir -p /home/dev
+RUN        chown dev:dev /home/dev
```
Comment (gustavonalle) [inline bin/docker/Dockerfile]: Maybe also collapse those 3 into a single ```RUN```, wdyt?

File: bin/docker/README.adoc [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+== The short version
+
+The short version. If you are lucky all you need is:
+
+[source]
+----
+# from the repository root
+> cd bin/docker
+> docker build -t infinispan/infinispan.github.io.git .
```
Comment (gustavonalle) [inline bin/docker/README.adoc]: should be ```docker build -t infinispan/infinispan.github.io .```

File: bin/docker/Dockerfile (hunk 2 above)
Comment (emmanuelbernard) [inline bin/docker/Dockerfile]: That's an interesting debate. I understand the demise of Docker caching but for that specific image maintained by non docker experts and used on the desktop (not in prod or anything), is it a good trade off to collapse these layers? I'm open to collapse but it has a bit of a readability cost.

Comment (gustavonalle) [inline bin/docker/Dockerfile]: The second line just installs ```wget```, so it actually does not help the readability

**Comment by gustavonalle** [comment]
@emmanuelbernard looks good!
Question: would that strategy work with a pre-build docker image (ideally as an automated build on https://hub.docker.com/) to avoid having to build it everytime? TBH It's a bit tedious to wait for it :smile: 
This could be a later enhancement though, not for this PR.

File: bin/docker/Dockerfile (hunk 2 above)
Comment (emmanuelbernard) [inline bin/docker/Dockerfile]: I'll do it but the wget for example is not required by the hibernate version of this.

**Comment by emmanuelbernard** [comment]
I have squasked a few Docker layers as per @gustavonalle 's comments.

@gustavonalle we could push it to Docker Hub. Caveat I am not familiar with that yet. The only thing that really chagrins me is that `rake setup` produces different outputs depending on `Gemfile.lock`, so I suspect that the same image being rebuild with a different Gemfil.lock (downloaded as part of git clone) would not be rebuilt if layer caching is enabled. But I guess that does not affect the Docker Hub vs manual build logic.

**Comment by tristantarrant** [comment]
Pushed, sir

**Comment by emmanuelbernard** [comment]
Thanks

**Pull request closed without merging**